packages/devtools-evm-hardhat/src/tasks/healthcheck/validate-rpcs.ts

```diff
@@ -0,0 +1,161 @@
+            }
+            ws.close()
+            resolve(success)
+        }
+
+        ws.onopen = () => {
+            const rpcRequest = JSON.stringify({
+                jsonrpc: JSON_RPC,
+                method: 'eth_blockNumber',
+                params: [],
+                id: 1,
+            })
+            ws.send(rpcRequest)
+        }
+
+        ws.onmessage = (event) => {
+            try {
+                const response = JSON.parse(event.data.toString())
+                if (response.jsonrpc === JSON_RPC && !!response.result) {
+                    cleanup(true)
+                } else {
+                    cleanup(false, `Invalid RPC response: ${event.data}`)
+                }
+            } catch (error) {
+                cleanup(false, `Error parsing RPC response: ${error}`)
+            }
+        }
+
+        ws.onerror = (error) => {
+            cleanup(false, `Validation failed: ${error instanceof Error ? error.message : 'An unknown error occurred'}`)
+        }
+
+        ws.onclose = () => {
+            cleanup(false, `Validation failed: WebSocket connection closed unexpectedly`)
+        }
+    })
+}
+
+const validateRpcUrl = async (rpcUrl: string | undefined, timeout: number, networkName: string): Promise<boolean> => {
+    if (!rpcUrl || rpcUrl.trim() === '') {
+        logger.error(`Missing RPC URL for network: ${networkName}`)
+        return false
+    }
+
+    logger.info(`...Validating RPC URL ${rpcUrl} for ${networkName}...`)
+
+    if (rpcUrl.startsWith(HTTP_URL) || rpcUrl.startsWith(HTTPS_URL)) {
+        return await validateHttpsRpcUrl(rpcUrl, timeout)
+    } else if (rpcUrl.startsWith(WS_URL) || rpcUrl.startsWith(WSS_URL)) {
+        return await validateWebSocketRpcUrl(rpcUrl, timeout)
+    }
+
+    logger.error(`Unsupported RPC protocol in network: ${networkName}`)
+
+    return false
+}
+
+const action: ActionType<TaskArguments> = async (taskArgs, hre) => {
+    printLogo()
+
+    const networks = hre.userConfig.networks || {}
+    const networkNames = Object.keys(networks)
+
+    logger.info(
+        `========== Validating RPC URLs with ${taskArgs.timeout}ms timeout for networks: ${networkNames.join(', ')}`
+    )
+
+    for (const networkName of networkNames) {
+        const rpcUrl = networks[networkName]?.[RPC_URL_KEY]
+
+        if (rpcUrl && !(await validateRpcUrl(rpcUrl, taskArgs.timeout, networkName))) {
+            throw new Error(`${printBoolean(false)} RPC URL validation failed for network: ${networkName}`)
+        }
+    }
+
+    logger.info(`${printBoolean(true)} All RPC URLs are valid!`)
+}
+task(
+    TASK_LZ_VALIDATE_RPCS,
+    'Validate RPC URLs in hardhat.config.ts. RPCs are only considered valid if they use the https or wss protocol and respond within the specified timeout.',
+    action
+).addParam(
+    'timeout',
+    `Maximum amount of time (in milliseconds) that the RPC URLs have to respond. If unspecified, default timeout will be used.`,
```

Comment:
`If unspecified, default timeout of ${TIMEOUT}ms will be used.`

Reply:
I had previously added this but noticed that when we run the task with `--help`, any defaults are automatically printed beside each argument:

<img width="1550" alt="Screenshot 2024-08-30 at 10 24 57 AM" src="https://github.com/user-attachments/assets/3e843b5a-2471-4912-b3dd-2a3d0d590612">

However, I understand where you are coming from and it is always good to be as clear as possible so I will add it to the description as well!